### ISSUE TITLE
StreamPeerBuffer is broken

### ISSUE THREAD

**Issue by raymoo** [godotengine/godot#10293]
**Operating system or device - Godot version:**
Debian GNU / Linux, Godot 3 alpha1

**Issue description:**
Putting data in a StreamPeerBuffer and reading it back out produces different results

**Steps to reproduce:**
Put this code in a node and run the scene:
```
extends Node

func _ready():
	if not test_read_write_equals_unsigned():
		print("Failed read write equals for unsigned ints")

func test_read_write_equals_unsigned():
	var buf = StreamPeerBuffer.new()
	buf.put_u8(69)
	buf.put_u16(420)
	buf.put_u32(123454321)
	
	var data = buf.get_data_array()
	var buf2 = StreamPeerBuffer.new()
	buf2.set_data_array(data)
	
	print(buf2.get_u8())
	print(buf2.get_u16())
	print(buf2.get_u32())
```

Output:
```
69
57088
65280
Failed read write equals for unsigned ints

```

**Comment by raymoo** [comment]
Note: The bug might actually be in StreamPeer, in which case it affects also StreamPeerTCP and StreamPeerSSL.

**Comment by raymoo** [comment]
Just kidding it's in StreamPeerBuffer, it's not updating or using the pointer in `get_partial_data`

**Comment by akien-mga** [comment]
Fixed by #10296.
